tests/test_upnp_client.py

```diff
@@ -101,6 +101,20 @@ async def test_init_bad_xml(self) -> None:
         with pytest.raises(UpnpXmlContentError):
             await factory.async_create_device("http://dlna_dmr:1234/device.xml")
 
+    @pytest.mark.asyncio
+    async def test_empty_descriptor(self) -> None:
```

Comment:
One more test to add would be `test_empty_descriptor_non_strict`, which looks like this test except with `factory = UpnpFactory(requester, non_strict=True)` and without the `pytest.raises` expectation.

See `test_bad_scpd_strict` and `test_bad_scpd_non_strict_fails` for contrasting tests.